packages/cli/src/LoadNodesAndCredentials.ts

```diff
@@ -299,7 +299,9 @@ export class LoadNodesAndCredentials {
 					sourcePath: path.join(directory, sourcePath),
 					supportedNodes:
 						loader instanceof PackageDirectoryLoader
-							? supportedNodes?.map((nodeName) => `${loader.packageName}.${nodeName}`)
+							? Object.keys(loader.nodeTypes).length > 0
```

Comment:
calling `Object.keys` over and over like this will be expensive, and it should be cached outside the loop.

Reply:
also, all this change is doing is adding **all** nodes from that loader on the `supportedNodes` array. This might fix the issue for you, but this is definitely not the fix, as it is likely to create other issues because now all credentials think that they can work with all node types.